app/views/submitted_content/_hyperlink.html.erb

```diff
@@ -1,146 +1,124 @@
-          if (timeout_flag) {
-            markEndTime();
-            while (windowsToClose!=null && windowsToClose.length>0){
-              windowsToClose.shift().close();
+        $(document).ready(function () {
+            let map_id = <%= @map.id %>; // map_id from response
+            let round = <%= @current_round %>; // round from response
+            let linksToUpdate = {}; // track links to restart timing on
+            let windowManager = new WindowManager()
+            const time_out_interval = 500; // in milliseconds
+
+            // Prevents right click on hyperlink
+            $('#url a.review_link').on("contextmenu", function (event) {
+                event.preventDefault(); // prevents default action i.e. does nothing on right click
+            });
+
+            // Prevents middle-click on hyperlink
+            $("#url a.review_link").on("auxclick", function (event) {
+                event.preventDefault(); // prevents default action i.e. does nothing on middle click
+            });
+
+            // Occurs on click of hyperlink
+            $("#url a.review_link").click(function linkAction(event) {
+                event.preventDefault(); // prevents defaults action of opening from occurring allowing our script to run
+                let link_handler = $(this); // current link
+                let address = link_handler.attr("href"); // url address of the current link
+
+                // If the address in already present in the array, prevent the user from opening it again
+                if (address in linksToUpdate) {
+                    if (linksToUpdate[address] === true) {
+                        return;
+                    }
+                }
+
+                let window_handler = window.open(address); // open link in new window/tab
+                windowManager.manage(window_handler); // track open windows
+
+                $("#restart_link_timing_flag").val('false'); // set restart link timing false
+                startTime(map_id, round, address); // save start time of link
+                linksToUpdate[address] = true; // add link to array to track and sets true
+
+                let link_timer = setInterval(checkWindowStatus, time_out_interval); // check window open/close every second
+
+                // Checks if window closed to then track end time
+                function checkWindowStatus() {
+                    let timeout_flag = $('#timeout_flag').val() == 'true'; // checks for timeout flag == true and if so sets
+                    if (window_handler.closed) { // if window closed
+                        endTime(map_id, round, address); // save end time of link
+                        linksToUpdate[address] = false; // sets link in tracking array to false
+                        clearInterval(link_timer); // removes check for window open/close
+                    }
+                    timeoutAction(timeout_flag); // if timeout flag is set, times out page
```

Comment:
Does this need to be inside the .click() call?

Reply:
Sadly, yes. It references the `window_handler` variable which is scoped to the that function.